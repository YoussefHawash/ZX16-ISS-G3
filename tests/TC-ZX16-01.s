.text
.org 0x0020

main:
    li x3, 5           # s0 = 5
    li x4, 5           # s1 = 5
    beq x3, x4, equal  #  branch
    li x5, 62          

equal:
    li x5, 42          # t1 = 42
    li x6, 0           # a0 = 0
    bz x6, done        #  branch

    li x5, 62          

done:
    ecall 0x008        # dump
    ecall 0x00A        # exit
