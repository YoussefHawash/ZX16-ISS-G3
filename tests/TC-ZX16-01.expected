LI x3, 0x05
LI x4, 0x05
BEQ x3, x4, 4
LI x5, 0x3E
LI x5, 0x2A
LI x6, 0x00
BZ x6, 4    
LI x5, 0x3E
ECALL 8    # Registers Dump
ECALL 10    # Program Exit